### ISSUE TITLE
`Mon` is `Ap`?

### ISSUE THREAD

**Issue by georgefst** [ekmett/reducers#28]
Isn't `Data.Semigroup.Monad.Mon` identical to `base`'s `Data.Monoid.Ap`?

Either way, it would be worth a haddock comment clarifying the relationship, and why `Mon` exists (just for backwards compatiblity?).

**Comment by RyanGlScott** [comment]
Yes, good observation. `Mon` is pretty ancient, dating back to 9b4450560018e6d2e876a6184737fa5a320998cd (before the `Applicative`–`Monad` proposal, IIRC). Now that `Applicative` is a superclass of `Monad`, there's not much reason to use `Mon` over `Ap`. I'd be happy to deprecate the former in favor of the latter.

Speaking of which, `reducers` has its own cargo-culted version of `Ap` [here](https://github.com/ekmett/reducers/blob/4125b912b9d9c6aa1950859c32c560235d608c52/src/Data/Semigroup/Applicative.hs#L63). Perhaps we should use the version from `base` on sufficiently recent versions of GHC.
